Diffusion/Common/Views/PromptTextField.swift

```diff
@@ -0,0 +1,152 @@
+        }
+        return nil
+    }
+    
+    private var packagesFilename: String? {
+        (filename as NSString?)?.deletingPathExtension
+    }
+    
+    private var compiledURL: URL? {
+        if let packagesFilename = packagesFilename {
+            return downloadedURL?.deletingLastPathComponent().appendingPathComponent(packagesFilename)
+        }
+        return nil
+    }
+    
+    private var textColor: Color {
+        switch tokenCount {
+        case 0...65:
+            return .green
+        case 66...75:
+            return .orange
+        default:
+            return .red
+        }
+    }
+    
+    // macOS initializer
+    init(text: Binding<String>, isPositivePrompt: Bool, model: Binding<String>) {
+         _textBinding = text
+         self.isPositivePrompt = isPositivePrompt
+        _model = model
+    }
+    
+    //iOS initializer
+    init(text: Binding<String>, isPositivePrompt: Bool, model: String) {
+        _textBinding = text
+        self.isPositivePrompt = isPositivePrompt
+        _model = .constant(model)
+    }
+
+    var body: some View {
+        VStack {
+            #if os(macOS)
+            TextField(isPositivePrompt ? "Positive prompt" : "Negative Prompt", text: $textBinding,
+                      axis: .vertical)
+                .lineLimit(20)
+                .textFieldStyle(.squareBorder)
+                .listRowInsets(EdgeInsets(top: 0, leading: -20, bottom: 0, trailing: 20))
+                .foregroundColor(textColor == .green ? .primary : textColor)
+                .frame(minHeight: 30)
+            if (modelInfo != nil && tokenizer != nil) {
+                HStack {
+                    Spacer()
+                    if !textBinding.isEmpty {
+                        Text("\(tokenCount)")
+                            .foregroundColor(textColor)
+                        Text(" / \(maxTokenCount)")
+                    }
+                }
+                .onReceive(Just(textBinding)) { text in
+                    updateTokenCount(newText: text)
+                }
+                .font(.caption)
+            }
+            #else
+            TextField("Prompt", text: $textBinding, axis: .vertical)
+                .lineLimit(20)
+                .listRowInsets(EdgeInsets(top: 0, leading: -20, bottom: 0, trailing: 20))
+                .foregroundColor(textColor == .green ? .primary : textColor)
+                .frame(minHeight: 30)
+            HStack {
+                if !textBinding.isEmpty {
+                    Text("\(tokenCount)")
+                        .foregroundColor(textColor)
+                    Text(" / \(maxTokenCount)")
+                }
+                Spacer()
+            }
+            .onReceive(Just(textBinding)) { text in
+                updateTokenCount(newText: text)
+            }
+            .font(.caption)
+            #endif
+        }
+        .onChange(of: model) { model in
+            updateTokenCount(newText: textBinding)
+        }
```

Comment:
```suggestion
        .onChange(of: model) { model in
            updateTokenCount(newText: textBinding)
        }
        .onAppear {
            updateTokenCount(newText: textBinding)
        }
```

I believe we have to call it once on app startup. Otherwise, the `tokenizer` will not be instantiated and therefore the `HStack` will not build the fields that update the count later. Without `.onAppear`, everything works _after_ you select a new model, but not if you keep the one used last time.

There might be other ways to achieve the same, but this looked easy. For example, unconditionally creating the `HStack` and the `.onReceive`, but then making sure that we do have a `tokenizer` when `updateTokenCount` is invoked.